Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -1336,15 +1336,27 @@ ShaderCode GeneratePixelShaderCode(APIType api_type, const ShaderHostConfig& hos
       {
         out.Write("\t\tcustom_data.final_color = float4(ocol0.x / 255.0, ocol0.y / 255.0, ocol0.z "
                   "/ 255.0, ocol0.w / 255.0);\n");
-        out.Write("\t\tfloat3 custom_output = {}_{}(custom_data).xyz;\n",
-                  CUSTOM_PIXELSHADER_COLOR_FUNC, i);
-        out.Write("\t\tocol0.xyz = uint3(custom_output.x * 255, custom_output.y * 255, "
-                  "custom_output.z * 255);\n");
+        out.Write("\t\tfloat4 custom_output = {}_{}(custom_data);\n", CUSTOM_PIXELSHADER_COLOR_FUNC,
+                  i);
+        out.Write("\t\tocol0 = uint3(custom_output.x * 255, custom_output.y * 255, "
+                  "custom_output.z * 255, custom_output.w * 255);\n");
+
+        if (use_dual_source)
+        {
+          out.Write("\t\tocol1.w = ocol0.w;\n");
+          out.Write("\t\tocol0.w = 255;\n");
+        }
```

Comment:
`WriteColor` doesn't handle `uint_output` and `use_dual_source` at the same time, which makes sense because `uint_output` is used for logic ops, and logic ops are disabled if blending is enabled, and `use_dual_source` is only ever enabled if blending is enabled (and dual source is supported). So there's no situation where both dual source and uint output are enabled at the same time.